### PR TITLE
Add findJavaHomePromise to export

### PR DESCRIPTION
**Pull request by Coder-Iro**

### CODE REVIEW

File: test.js [hunk 1]
```diff
@@ -16,7 +16,7 @@
 'use strict';
 
 describe('find-java-home', function(){
-  var sut = require('./');
+  var sut = require('./').findJavaHome;
```
Comment (jsdevel) [inline test.js:19]: yea, we're not going to change the export interface for this.  can you assign findJavaHomePromise as a property to findJavaHome?  That way you can do `require('find-java-home').findJavaHomePromise`.  Or maybe this is better? `require('find-java-home').promise`

**Review by jsdevel — CHANGES_REQUESTED** [review]

**Comment by Coder-Iro** [comment]
Travis check is failing because node 8 is too low

**Comment by Coder-Iro** [comment]
@jsdevel PLZ merge this PR ASAP. I have to use this module in my project.

File: package.json [hunk 2]
```diff
@@ -33,7 +33,7 @@
     "@types/node": "^12.6.8",
     "@types/which": "^1.3.1",
     "@types/winreg": "^1.2.30",
-    "mocha": "~1.18.2",
```
[the remark this answers is not in the document]
Comment (Coder-Iro) [reply package.json]: I reverted that change.

**Review by jsdevel — CHANGES_REQUESTED** [review]
sorry for the late re-review.  can you completely remove package-lock.json changes?

**Comment by Coder-Iro** [comment]
When will this change be distributed to the npm registry?

**Comment by jsdevel** [comment]
@Coder-Iro it should be in 1.2.0

**Comment by Coder-Iro** [comment]
I think there should be another example for Promise in README.

**Comment by jsdevel** [comment]
@Coder-Iro please submit a pr!

**Pull request merged**